File: src/main/java/com/fishiphedia/ranking/service/RankingService.java
package com.fishiphedia.ranking.service;

import com.fishiphedia.ranking.dto.FisherRankingResponse;
import com.fishiphedia.ranking.dto.FishCollectionRankingResponse;
import java.util.List;

public interface RankingService {
    List<FisherRankingResponse> getFisherRanking();
    List<FishCollectionRankingResponse> getFishCollectionRanking();
    List<FishCollectionRankingResponse> getFishRankingAllFish();
    List<FishCollectionRankingResponse> getFishRankingByFish(Long fishId);
} 

File: frontend/src/services/rankingService.ts
import api from './api';

export interface FisherRanking {
  userId: number;
  name: string;
  totalScore: number;
}

export interface FishCollectionRanking {
  userId: number;
  name: string;
  fishId: number;
  fishName: string;
  totalScore: number;
  highestScore: number;
  highestLength: number;
}

export const rankingService = {
  // 낚시꾼 전체 랭킹
  async getFisherRanking(): Promise<FisherRanking[]> {
    const res = await api.get('/ranking/fisher');
    return res.data;
  },
  // FishCollection 전체 랭킹
  async getFishCollectionRanking(): Promise<FishCollectionRanking[]> {
    const res = await api.get('/ranking/fish-collection');
    return res.data;
  },
  // 모든 물고기별 랭킹
  async getFishRankingAllFish(): Promise<FishCollectionRanking[]> {
    const res = await api.get('/ranking/fish');
    return res.data;
  },
  // 특정 물고기별 랭킹
  async getFishRankingByFish(fishId: number): Promise<FishCollectionRanking[]> {
    const res = await api.get(`/ranking/fish/${fishId}`);
    return res.data;
  },
}; 